refactor: refactor source.py

kzamlynska opened the pull request:
extract sources to separate files

mhordynski reviewed (changes requested):
imo we should create package `ragbits.document_search.documents.sources` and there create separate modules per source type.

Preferably to not change public API let's expose all the sources in init.py making sure that correct extras are in place.

kzamlynska commented:
> imo we should create package `ragbits.document_search.documents.sources` and there create separate modules per source type.
> 
> Preferably to not change public API let's expose all the sources in init.py making sure that correct extras are in place.

please check my changes

micpst reviewed (commented):
looks better, but please keep naming consistent with the rest of the lib. should be:
- sources/base.py (we can also put source resolver here)
- sources/gcs.py
- sources/hf.py
- source/local.py

kzamlynska commented:
> looks better, but please keep naming consistent with the rest of the lib. should be:
> 
> * sources/base.py (we can also put source resolver here)
> * sources/gcs.py
> * sources/hf.py
> * sources/local.py

I've changed the names of files and I've moved the source resolver to the base.py file

kzamlynska commented on packages/ragbits-document-search/src/ragbits/document_search/documents/sources/base.py:
```diff
@@ -0,0 +1,199 @@
+from pydantic_core import CoreSchema, core_schema
+
+with suppress(ImportError):
+    pass
```
removed

kzamlynska commented on packages/ragbits-document-search/src/ragbits/document_search/documents/sources/base.py:
```diff
@@ -0,0 +1,199 @@
+    pass
+
+with suppress(ImportError):
+    pass
```
removed

ludwiktrammer commented on docs/how-to/document_search/search_documents.md:
```diff
@@ -14,7 +15,8 @@ from ragbits.core.embeddings.litellm import LiteLLMEmbeddings
 from ragbits.document_search import DocumentSearch
 from ragbits.document_search.documents.document import DocumentMeta
-from ragbits.document_search.documents.sources import GCSSource
+from ragbits.document_search.documents.sources.gcs import GCSSource
```
Cann't we just export the various sources from the main `ragbits.document_search.documents.sources` package? (i.e., the `__init__.py` file there)

mhordynski commented on examples/evaluation/document-search/basic/optimize.py:
```diff
@@ -59,7 +59,7 @@
                 },
                 "source": {
-                    "type": "ragbits.document_search.documents.sources:HuggingFaceSource",
+                    "type": "ragbits.document_search.documents.hf:HuggingFaceSource",
```
I think that's wrong. Shouldn't be `ragbits.document_search.documents.sources`?

kzamlynska replied:
Yes, thanks for pointing it.

mhordynski commented on examples/evaluation/document-search/basic/evaluate.py:
```diff
@@ -57,7 +57,7 @@
             },
             "source": {
-                "type": "ragbits.document_search.documents.sources:HuggingFaceSource",
+                "type": "ragbits.document_search.documents.hf:HuggingFaceSource",
```
documents.hf -> documents.sources

kzamlynska replied:
changed.

mhordynski reviewed (approved):
lgtm
